leg_detector/src/leg_detector.cpp

```diff
@@ -725,9 +725,9 @@ class LegDetector
       vector<float> f = calcLegFeatures(*i, *scan);
 
       for (int k = 0; k < feat_count_; k++)
-        tmp_mat->data.fl[k] = (float)(f[k]);
+        tmp_mat.data[k] = (float)(f[k]);
 
-      float probability = forest.predict_prob(tmp_mat);
+      float probability = 0.5 - forest->predict(tmp_mat, cv::noArray(), cv::ml::RTrees::PREDICT_SUM) / forest->getRoots().size();
```

Comment:
Can `forest->getRoots().size()` be `0`? Then check for it, and set `probability = 0` in that case.